content/WebAuthn/Concepts/Using_WebAuthn_for_Signing.adoc

```diff
@@ -0,0 +1,25 @@
+
+== High-Level Architecture
+This section will discuss at a high level the architecture necessary for adapting a WebAuthn flow for file signing. The actual implementation will depend on the environment, WebAuthn server or library and the WebAuthn client (brower or local) in use, and is outside of the scope of this document.
+
+The method for leveraging WebAuthn for signature operations on arbitrary files is built around the FIDO Signature Creation flow, comprised of the link:https://www.w3.org/TR/webauthn-2/#sctn-op-make-cred[MakeCredential] and link:https://www.w3.org/TR/webauthn-2/#sctn-op-get-assertion[GetAssertion] operations.
+
+In the link:https://www.w3.org/TR/webauthn-2/#sctn-op-make-cred[MakeCredential] operation, the challenge is defined as a random string. To ensure each credential and keypair generated is unique, this challenge should remain unique for the MakeCredential operation. However, in the link:https://www.w3.org/TR/webauthn-2/#sctn-op-make-cred[MakeCredential] operation, the algorithm for the keypair must be defined. The WebAuthn framework defines the algorithm to be used when the authentication generates the keypair via the identifier registered in the IANA COSE Algorithm Registry, as detailed in the link:https://www.w3.org/TR/webauthn-2/#sctn-alg-identifier[Cryptographic Algorithm Identifier] section. For example, EdDSA is defined as (-8). The requirements for detailing the algorithm to be used is well defined in the WebAuthn specification, however not all authenticators support all algorithms. It is recommended to include logic to capture the case where a user attempts to use an algorithm not supported by their authenticator.
+
+Further, the link:https://www.w3.org/TR/webauthn-2/#enum-userVerificationRequirement[User Verification] behavior is also defined by the MakeCredential operation, allowing a relying party to define if the created credential will require a PIN or biometric supplied by the user to access. This can allow for securing keypairs used for signing to only valid users of the WebAuthn Authenticator. In addition, it is also recommneded link:https://www.w3.org/TR/webauthn-2/#sctn-attestation[Attestation] is enabled during the registration event to verify the authenticity of the WebAuthn authenticator being used. Other fields in the MakeCredential can be provided with other identifying information or left to default values. However, it is important to note the orginID should match the endpoint the user is provided to access the signing service, as is the case in a standard WebAuthn service.
```

Comment:
Typo:
```suggestion
Further, the link:https://www.w3.org/TR/webauthn-2/#enum-userVerificationRequirement[User Verification] behavior is also defined by the MakeCredential operation, allowing a relying party to define if the created credential will require a PIN or biometric supplied by the user to access. This can allow for securing keypairs used for signing to only valid users of the WebAuthn Authenticator. In addition, it is also recommneded link:https://www.w3.org/TR/webauthn-2/#sctn-attestation[Attestation] is enabled during the registration event to verify the authenticity of the WebAuthn authenticator being used. Other fields in the MakeCredential can be provided with other identifying information or left to default values. However, it is important to note the originID should match the endpoint the user is provided to access the signing service, as is the case in a standard WebAuthn service.
```
But the "originID" term is confusing to me, it's not clear what it refers to. Does it mean the `rp.id` parameter or the `clientDataJSON.origin` response? Or something else, even?